auction-app-frontend/src/components/CheckoutPaymentForm/CheckoutPaymentForm.jsx

```diff
@@ -0,0 +1,120 @@
+import { useState } from "react";
+import { useStripe, useElements, CardNumberElement, CardExpiryElement, CardCvcElement } from '@stripe/react-stripe-js';
+
+import { LoadingComponent, Button, ErrorComponent } from "src/components";
+import { BUTTON_VARIANTS, BUTTON_LABELS } from "src/constants";
+
+import "./style.scss";
+
+const CheckoutPaymentForm = ({ clientSecret, onPaymentSuccess }) =>
+{
+    const [errorMessage, setErrorMessage] = useState("");
+    const [successMessage, setSuccessMessage] = useState("");
+    const [cardNumberError, setCardNumberError] = useState("");
+    const [cardExpiryError, setCardExpiryError] = useState("");
+    const [cardCvcError, setCardCvcError] = useState("");
+    const [cardholderName, setCardholderName] = useState("");
+
+    const stripe = useStripe();
+    const elements = useElements();
+
+    const handleSubmit = async (event) =>
+    {
+        event.preventDefault();
+
+        if (!stripe || !elements) {
+            setErrorMessage("Stripe has not initialized");
+            return;
+        }
+
+        const cardElement = elements.getElement(CardNumberElement);
+
+        if (!cardElement) {
+            setErrorMessage("Card details not available");
+            return;
+        }
+
+        try {
+            const { token, error } = await stripe.createToken(cardElement, { name: cardholderName });
+            if (error) {
```

Comment:
Why do we need this, if there is an error, shouldn't that be handled by the catch block>